test/hint.lean

```diff
@@ -4,14 +4,24 @@ import Mathlib.Data.Nat.Prime
 
 /--
 info: Try these:
-• linarith
+• simp_all only
```

Comment:
What's the explanation for this change? Does `simp_all only` win a race? Is it deterministic?
(Same question for the other changes below, too.)

Reply:
yes, `simp_all` is faster. These tests are indeed nondeterministic now, and if that becomes a problem we'll have to remove them (or perhaps set an option to disable parallelism?)

I've compiled this file many times locally and the guard conditions haven't failed so far.

Reply:
In CI, of course, the non-determinism showed up. I've removed some of the `guard_msg` conditions.